src/overlays/actors/ovl_En_Dragon/z_en_dragon.c

```diff
@@ -37,130 +72,777 @@ const ActorInit En_Dragon_InitVars = {
+            gSaveContext.save.weekEventReg[0xD] |= 1;
+            switch (this->pythonIndex) {
+                case 0:
+                    gSaveContext.save.weekEventReg[0x53] |= 0x10;
+                    break;
+
+                case 1:
+                    gSaveContext.save.weekEventReg[0x53] |= 0x20;
+                    break;
+
+                case 2:
+                    gSaveContext.save.weekEventReg[0x53] |= 0x40;
+                    break;
+
+                case 3:
+                    gSaveContext.save.weekEventReg[0x53] |= 0x80;
+                    break;
+                case 4:
+                    gSaveContext.save.weekEventReg[0x54] |= 1;
+                    break;
+
+                case 5:
+                    gSaveContext.save.weekEventReg[0x54] |= 2;
+                    break;
+
+                case 6:
+                    gSaveContext.save.weekEventReg[0x54] |= 4;
+                    break;
+
+                case 7:
+                    gSaveContext.save.weekEventReg[0x54] |= 8;
+                    break;
+            }
+        }
+    }
+
+    Actor_MarkForDeath(&this->actor);
+}
+
+void EnDragon_UpdateDamage(EnDragon* this, GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
+    u32 sp30;
+
+    if ((this->action == DEEP_PYTHON_ACTION_EXTEND) && ((this->collider.elements[2].info.bumperFlags & BUMP_HIT) ||
+                                                        (this->collider.elements[3].info.bumperFlags & BUMP_HIT) ||
+                                                        (this->collider.elements[4].info.bumperFlags & BUMP_HIT) ||
+                                                        (this->collider.elements[5].info.bumperFlags & BUMP_HIT) ||
+                                                        (this->collider.elements[6].info.bumperFlags & BUMP_HIT) ||
+                                                        (this->collider.elements[7].info.bumperFlags & BUMP_HIT))) {
+        Actor_ApplyDamage(&this->actor);
+        Actor_SetColorFilter(&this->actor, 0x4000, 255, 0, 25);
+        if (this->actor.colChkInfo.health > 0) {
+            Actor_PlaySfxAtPos(&this->actor, NA_SE_EN_UTSUBO_DAMAGE);
+            this->action = DEEP_PYTHON_ACTION_DAMAGE;
+        } else {
+            Enemy_StartFinishingBlow(globalCtx, &this->actor);
+            Actor_PlaySfxAtPos(&this->actor, NA_SE_EN_UTSUBO_DEAD);
+            this->actor.flags |= ACTOR_FLAG_8000000;
+            this->actor.flags &= ~ACTOR_FLAG_1;
+            this->actor.flags |= ACTOR_FLAG_100000;
+            this->action = DEEP_PYTHON_ACTION_SETUP_DEAD;
+            this->actionFunc = EnDragon_SetupDead;
+        }
+    }
+
+    if ((this->action == DEEP_PYTHON_ACTION_EXTEND) && (this->grabWaitTimer == 0) &&
+        (player->invincibilityTimer == 0) && (this->collider.elements[0].info.ocElemFlags & OCELEM_HIT) &&
+        (!(func_800B64FC(globalCtx, 1000.0f, &this->actor.world.pos, &sp30) >= 0.0f) || (sp30 != 1))) {
```

Comment:
This line has to be written like this? a `<` wouldn't work?
```suggestion
        ((func_800B64FC(globalCtx, 1000.0f, &this->actor.world.pos, &sp30) < 0.0f) || (sp30 != 1))) {
```

Reply:
Correct, it has to be written like this, as silly as it looks:
![image](https://user-images.githubusercontent.com/12245827/160677938-b9087d58-9e0b-42a7-a54a-7a2fb5012d00.png)